### PR TITLE
Adds SDK + CLI commands for enumeration Runs/Actions

### PR DESCRIPTION
**Pull request by urimandujano**

### CODE REVIEW

File: globus_automate_client/cli/flows.py [hunk 1]
```diff
@@ -918,6 +919,77 @@ def flow_action_log(
             format_and_echo(resp, verbose=verbose)
 
 
+@app.command("action-enumerate")
+def flow_action_enumerate(
+    roles: List[ActionRole] = typer.Option(
+        [],
+        "--role",
+        help="Display Actions where you have the selected role. [repeatable]",
```
Comment (jimPruyne) [inline globus_automate_client/cli/flows.py:927]: I assume the `ActionRole` enumeration provides the help on the allowed values?

File: globus_automate_client/flows_client.py [hunk 2]
```diff
@@ -593,6 +593,76 @@ def flow_action_cancel(
...
+            Flow listing. The keys indicate the field to order on, and
+            the value can be either ASC, for ascending order, or DESC, for
+            descending order. The first ordering criteria will be used to sort
+            the data, subsequent ordering criteria will be applied for ties.
+            Note: To ensure orderings are applied in the correct order, use an
+            OrderedDict if trying to apply multiple orderings.
+        """
+        params = {}
+        if roles is not None and len(roles) > 0:
+            params.update(dict(filter_roles=",".join(roles)))
```
Comment (jimPruyne) [inline globus_automate_client/flows_client.py:650]: Any reason not to use:
```python
params['filter_roles'] = ",".join(roles)
```

**Review by jimPruyne — COMMENTED** [review]
One minor clarification. One minor change idea

File: globus_automate_client/cli/flows.py (hunk 1 above)
Comment (urimandujano) [reply globus_automate_client/cli/flows.py:927]: That's 100% correct:

```
  --role [created_by|monitor_by|manage_by]
                                  Display Actions where you have the selected
                                  role. [repeatable]  [default: ]
```

File: globus_automate_client/flows_client.py (hunk 2 above)
Comment (urimandujano) [reply globus_automate_client/flows_client.py]: That's a good one!

**Comment by urimandujano** [comment]
@jimPruyne finally made a round of updates on this. Any further thoughts?

**Comment by urimandujano** [comment]
Users wanting to simply provide their own http_timeouts on clients were unable to use the `create_flows_client` helper method and had to resort to custom creating clients via FlowsClient.new_client() which involved distinct package imports and identifying valid defaults.

This PR allows full customization of the underlying FlowsClient via keyword-only arguments to `create_flows_client`.

**Review by jimPruyne — APPROVED** [review]

**Pull request merged**